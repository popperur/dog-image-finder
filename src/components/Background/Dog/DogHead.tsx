import {
  DogCollar,
  DogEyeLeft,
  DogEyeRight,
  DogEyesContainer,
  DogFace,
  DogFaceWhite,
  DogHeadContainer,
  DogHeadLeftEar,
  DogHeadRightEar,
  DogNose,
} from "components/Background/Dog/styles.tsx";
import useAnimationTimer from "hooks/useAnimationTimer.ts";

function DogHead() {
  const [keyIndex, animationDuration] = useAnimationTimer({
    initial: { duration: 2, delay: 4 },
    durationFn: () => 1.5 + Math.random() * 4,
    delayFn: () => 2 + Math.random() * 14,
  });

  return (
    <DogHeadContainer>
      <DogHeadLeftEar />
      <DogHeadRightEar />
      <DogEyesContainer
        $duration={animationDuration}
        key={`dog-eyes-container-${keyIndex}`}
      >
        <DogEyeLeft />
        <DogEyeRight />
      </DogEyesContainer>
      <DogFace>
        <DogFaceWhite
          $duration={animationDuration}
          key={`dog-face-white-${keyIndex}`}
        >
          <DogNose $duration={animationDuration} key={`dog-nose-${keyIndex}`} />
        </DogFaceWhite>
      </DogFace>
      <DogCollar />
    </DogHeadContainer>
  );
}

export default DogHead;
