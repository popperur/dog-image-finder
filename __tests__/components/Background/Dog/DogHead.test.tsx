import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import DogHead from "components/Background/Dog/DogHead";

describe("DogHead component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<DogHead />);

    expect(asFragment()).toMatchSnapshot();
  });
});
