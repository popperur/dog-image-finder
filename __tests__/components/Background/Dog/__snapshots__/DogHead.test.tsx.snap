// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DogHead component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  right: 0;
  bottom: 0;
  width: 100px;
  height: 100%;
  background-color: #ff9a29;
  border-top-right-radius: 60px;
  border-top-left-radius: 60px;
  border-bottom-right-radius: 120px;
}

.c1 {
  position: absolute;
  top: -18px;
  width: 25px;
  height: 80px;
  background-color: #ff9a29;
}

.c2 {
  left: 0;
  border-top-right-radius: 40px;
}

.c3 {
  right: 0;
  border-top-left-radius: 40px;
}

.c4 {
  position: absolute;
  animation-duration: 2s;
  animation-timing-function: cubic-bezier(1, 0, 0, 1);
}

.c5 {
  top: 25px;
  right: -2px;
  width: 100%;
  height: 9px;
  transform: rotatex(9deg);
  animation-name: bylGws;
}

.c6 {
  position: absolute;
  width: 12px;
  height: 100%;
  background-color: #111;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

.c7 {
  left: 30px;
}

.c8 {
  left: 75px;
}

.c9 {
  position: absolute;
  top: 44px;
  width: 100%;
  height: 42px;
}

.c10 {
  right: -10px;
  width: 65px;
  height: 100%;
  background: #fff7ee;
  border-radius: 20px 20px 42px 38px;
  animation-name: jZKbdW;
}

.c11 {
  right: 0;
  top: 0;
  width: 17px;
  height: 14px;
  background: #000;
  border-radius: 0 0 0 38px;
  animation-name: cnrYMx;
}

.c12 {
  position: absolute;
  top: 92px;
  width: 100%;
  height: 10px;
  background-color: #c64f3c;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    />
    <div
      class="c1 c3"
    />
    <div
      class="c4 c5"
    >
      <div
        class="c6 c7"
      />
      <div
        class="c6 c8"
      />
    </div>
    <div
      class="c9"
    >
      <div
        class="c4 c10"
      >
        <div
          class="c4 c11"
        />
      </div>
    </div>
    <div
      class="c12"
    />
  </div>
</DocumentFragment>
`;
